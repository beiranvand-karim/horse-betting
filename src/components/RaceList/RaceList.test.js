/* global describe, it, jest, expect*/
import {shallow} from "enzyme"
import {RaceList} from "./"
import React from "react"
import renderer from "react-test-renderer"
import {MemoryRouter} from "react-router-dom"
import game from "../../backend/game"

describe('<RaceList/>', () => {

   const races = game.races;

   it('should render without crashing', () => {
      shallow(<RaceList raceList={races}/>)
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><RaceList raceList={races}/></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });
});
