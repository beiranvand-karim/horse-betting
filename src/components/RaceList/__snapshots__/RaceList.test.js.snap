// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RaceList/> should matches the snapshot 1`] = `
<div>
  <h3>
    race list
  </h3>
  <div>
    <h4>
      race number: 
      3
    </h4>
    <h4>
      race startTime: 
      2019-04-04T10:03:38
    </h4>
    <div>
      <h5>
        start number: 
        1
      </h5>
      <h5>
        horse name: 
        Lani Kai Beach
      </h5>
      <h5>
        driver first name: 
        J N
         
        Douglass
      </h5>
      <h5>
        horse father: 
        Camlach
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        2
      </h5>
      <h5>
        horse name: 
        The Bellum
      </h5>
      <h5>
        driver first name: 
        Renata
         
        Furina
      </h5>
      <h5>
        horse father: 
        Union Guy
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        3
      </h5>
      <h5>
        horse name: 
        Luke By Knight
      </h5>
      <h5>
        driver first name: 
        Lachie
         
        Manzelmann
      </h5>
      <h5>
        horse father: 
        Live Or Die
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        4
      </h5>
      <h5>
        horse name: 
        Clementsorrell
      </h5>
      <h5>
        driver first name: 
        W H
         
        Rixon
      </h5>
      <h5>
        horse father: 
        Totally Ruthless
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        5
      </h5>
      <h5>
        horse name: 
        Vinny Gambini
      </h5>
      <h5>
        driver first name: 
        T P
         
        Mccarthy
      </h5>
      <h5>
        horse father: 
        Vintage Master
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        6
      </h5>
      <h5>
        horse name: 
        All Positive
      </h5>
      <h5>
        driver first name: 
        D R
         
        Morris
      </h5>
      <h5>
        horse father: 
        Armbro Operative
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        7
      </h5>
      <h5>
        horse name: 
        The Pix
      </h5>
      <h5>
        driver first name: 
        G W
         
        Mcelhinney
      </h5>
      <h5>
        horse father: 
        Modern Art
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        8
      </h5>
      <h5>
        horse name: 
        Sals Luck
      </h5>
      <h5>
        driver first name: 
        Cameron
         
        Hart
      </h5>
      <h5>
        horse father: 
        Ifihadyourluck
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        9
      </h5>
      <h5>
        horse name: 
        Art I Special
      </h5>
      <h5>
        driver first name: 
        W H
         
        Rixon
      </h5>
      <h5>
        horse father: 
        Artsplace
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        10
      </h5>
      <h5>
        horse name: 
        Bara Jayla
      </h5>
      <h5>
        driver first name: 
        M B
         
        Towers
      </h5>
      <h5>
        horse father: 
        Santanna Blue Chip
      </h5>
    </div>
  </div>
  <div>
    <h4>
      race number: 
      4
    </h4>
    <h4>
      race startTime: 
      2019-04-04T10:34:09
    </h4>
    <div>
      <h5>
        start number: 
        1
      </h5>
      <h5>
        horse name: 
        Simply Clever
      </h5>
      <h5>
        driver first name: 
        Cameron
         
        Hart
      </h5>
      <h5>
        horse father: 
        Santanna Blue Chip
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        2
      </h5>
      <h5>
        horse name: 
        Aintnobettor
      </h5>
      <h5>
        driver first name: 
        I L
         
        Ross
      </h5>
      <h5>
        horse father: 
        Bettors Delight
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        3
      </h5>
      <h5>
        horse name: 
        Little Change
      </h5>
      <h5>
        driver first name: 
        R J
         
        Alchin
      </h5>
      <h5>
        horse father: 
        Changeover
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        4
      </h5>
      <h5>
        horse name: 
        Sam Is Perfection
      </h5>
      <h5>
        driver first name: 
        G J
         
        Fitzpatrick
      </h5>
      <h5>
        horse father: 
        Pay Me Christian
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        5
      </h5>
      <h5>
        horse name: 
        The Maldives
      </h5>
      <h5>
        driver first name: 
        J N
         
        Douglass
      </h5>
      <h5>
        horse father: 
        Tinted Cloud
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        6
      </h5>
      <h5>
        horse name: 
        Burning Rubber
      </h5>
      <h5>
        driver first name: 
        J E
         
        Trainor
      </h5>
      <h5>
        horse father: 
        Mach Three
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        7
      </h5>
      <h5>
        horse name: 
        Persimmon
      </h5>
      <h5>
        driver first name: 
        T P
         
        Mccarthy
      </h5>
      <h5>
        horse father: 
        Courage Under Fire
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        8
      </h5>
      <h5>
        horse name: 
        Bettor To Be Tricky
      </h5>
      <h5>
        driver first name: 
        K A
         
        Turner
      </h5>
      <h5>
        horse father: 
        Bettor'S Delight
      </h5>
    </div>
  </div>
  <div>
    <h4>
      race number: 
      5
    </h4>
    <h4>
      race startTime: 
      2019-04-04T11:07:48
    </h4>
    <div>
      <h5>
        start number: 
        1
      </h5>
      <h5>
        horse name: 
        Shezsavvy
      </h5>
      <h5>
        driver first name: 
        Ellen
         
        Rixon
      </h5>
      <h5>
        horse father: 
        Betterthancheddar
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        2
      </h5>
      <h5>
        horse name: 
        Courage Lips
      </h5>
      <h5>
        driver first name: 
        A J
         
        Turnbull
      </h5>
      <h5>
        horse father: 
        Courage Under Fire
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        3
      </h5>
      <h5>
        horse name: 
        Bara Halley
      </h5>
      <h5>
        driver first name: 
        M B
         
        Towers
      </h5>
      <h5>
        horse father: 
        Mister Big
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        4
      </h5>
      <h5>
        horse name: 
        Semi Sensation
      </h5>
      <h5>
        driver first name: 
        D J
         
        Lindon
      </h5>
      <h5>
        horse father: 
        Mr Feelgood
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        5
      </h5>
      <h5>
        horse name: 
        King Fisher
      </h5>
      <h5>
        driver first name: 
        G W
         
        Mcelhinney
      </h5>
      <h5>
        horse father: 
        Mr Feelgood
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        6
      </h5>
      <h5>
        horse name: 
        Modern Metropolis
      </h5>
      <h5>
        driver first name: 
        T P
         
        Mccarthy
      </h5>
      <h5>
        horse father: 
        Metropolitan
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        7
      </h5>
      <h5>
        horse name: 
        No Risk No Return
      </h5>
      <h5>
        driver first name: 
        Jack
         
        Wade
      </h5>
      <h5>
        horse father: 
        Life Sign
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        8
      </h5>
      <h5>
        horse name: 
        The Maniac
      </h5>
      <h5>
        driver first name: 
        P
         
        Breaden
      </h5>
      <h5>
        horse father: 
        Armbro Operative
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        9
      </h5>
      <h5>
        horse name: 
        Pure Charisma
      </h5>
      <h5>
        driver first name: 
        J N
         
        Douglass
      </h5>
      <h5>
        horse father: 
        Always A Virgin
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        10
      </h5>
      <h5>
        horse name: 
        Diamond Thief
      </h5>
      <h5>
        driver first name: 
        Lachie
         
        Manzelmann
      </h5>
      <h5>
        horse father: 
        Speed King
      </h5>
    </div>
  </div>
  <div>
    <h4>
      race number: 
      6
    </h4>
    <h4>
      race startTime: 
      2019-04-04T11:35:48
    </h4>
    <div>
      <h5>
        start number: 
        1
      </h5>
      <h5>
        horse name: 
        Sputnik
      </h5>
      <h5>
        driver first name: 
        J J
         
        Reynolds
      </h5>
      <h5>
        horse father: 
        Mr Feelgood
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        2
      </h5>
      <h5>
        horse name: 
        Bold Ambition
      </h5>
      <h5>
        driver first name: 
        J N
         
        Douglass
      </h5>
      <h5>
        horse father: 
        Pet Rock
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        3
      </h5>
      <h5>
        horse name: 
        Colossal Stride
      </h5>
      <h5>
        driver first name: 
        Lachie
         
        Manzelmann
      </h5>
      <h5>
        horse father: 
        Bettors Delight
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        4
      </h5>
      <h5>
        horse name: 
        Cannan
      </h5>
      <h5>
        driver first name: 
        A M
         
        Butt
      </h5>
      <h5>
        horse father: 
        Sportswriter
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        5
      </h5>
      <h5>
        horse name: 
        Takinover
      </h5>
      <h5>
        driver first name: 
        T P
         
        Mccarthy
      </h5>
      <h5>
        horse father: 
        Blissfull Hall
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        6
      </h5>
      <h5>
        horse name: 
        Jackbkwik
      </h5>
      <h5>
        driver first name: 
        R J
         
        Munnoch
      </h5>
      <h5>
        horse father: 
        Smiling Shard
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        7
      </h5>
      <h5>
        horse name: 
        Devouged
      </h5>
      <h5>
        driver first name: 
        B J
         
        Abbott
      </h5>
      <h5>
        horse father: 
        Roll With Joe
      </h5>
    </div>
    <div>
      <h5>
        start number: 
        8
      </h5>
      <h5>
        horse name: 
        Humming Top
      </h5>
      <h5>
        driver first name: 
        R W
         
        Adams
      </h5>
      <h5>
        horse father: 
        Always A Virgin
      </h5>
    </div>
  </div>
</div>
`;
